site/docs/uk/hosting/heroku.md

```diff
@@ -0,0 +1,424 @@
+    supports_inline_queries: false,
+  },
+});
+```
+
+:::
+
+Круто!
+Настала пора підготувати наше середовище для розгортання!
+Усі прямуємо до [розділу про розгортання](#розгортання)! :muscle:
+
+## Тривале опитування
+
+::: warning Ваш скрипт буде працювати безперервно при використанні тривалого опитування.
+Якщо ви не знаєте, як керувати цією поведінкою, переконайтеся, що у вас є достатньо [dyno hours](https://devcenter.heroku.com/articles/eco-dyno-hours) для цього.
+:::
+
+> Думаєте про використання вебхуків?
+> Перейдіть до [розділу про вебхуки](#вебхуки). :rocket:
+
+Використання тривалого опитування на вашому сервері не завжди є поганою ідеєю.
+Іноді воно підходить для ботів, що збирають дані і не потребують швидкої відповіді та обробки великої кількості даних.
+Якщо ви хочете робити це раз на годину, це легко зробити.
+Але цього не можна контролювати за допомогою вебхуків.
+Якщо ваш бот буде завалений повідомленнями, вам прийде багато запитів на вебхук, проте за допомогою тривалого опитування ви можете легко обмежити швидкість обробки оновлень.
+
+### Створення `bot.ts`
+
+Давайте відкриємо файл `bot.ts`, який ми створили раніше.
+Він має містити наступний код:
+
+```ts
+import { Bot } from "grammy";
+
+const token = process.env.BOT_TOKEN;
+if (!token) throw new Error("BOT_TOKEN не вказано");
+
+const bot = new Bot(token);
+
+bot.command(
+  "start",
+  (ctx) =>
+    ctx.reply("Я запущений на Heroku з використанням тривалого опитування!"),
+);
+
+bot.start();
+```
+
+Ось і все!
+Ми готові розгорнути його.
+Досить просто, чи не так? :smiley:
+Якщо ви думаєте, що це занадто просто, перевірте наш [контрольний список розгортання](../advanced/deployment.md#тривале-опитування)! :rocket:
+
+## Розгортання
+
+Ні... ваш _Rocket Bot_ ще не готовий до запуску.
+Спочатку завершіть наступні етапи!
+
+### Компіляція файлів
+
+Запустіть цей код у терміналі, щоб скомпілювати файли TypeScript в JavaScript:
+
+```sh
+npx tsc
+```
+
+Якщо він успішно виконається й не видасть жодних помилок, скомпільовані файли з розширенням `.js` повинні знаходитись у каталозі `dist`.
+
+### Налаштування `Procfile`
+
+Наразі у `Heroku` є кілька [типів dyno](https://devcenter.heroku.com/articles/dyno-types).
+Два з них:
+
+- **Веб dyno**:
+  <br> _Веб dyno_ процесу "web", які отримують HTTP трафік від маршрутизаторів.
+  Цей тип dyno має 30-ти секундний тайм-аут для виконання коду.
+  Крім того, він засинає, якщо протягом 30-ти хвилин не було запиту для обробки.
+  Цей тип dyno цілком підходить для _вебхуків_.
+
+- **Worker dyno**:
+  <br> _Worker dyno_ зазвичай використовуються для фонових задач.
+  Він НЕ має тайм-ауту і НЕ засинає, якщо не обробляє жодного вебзапиту.
+  Він підходить для _тривалого опитування_.
+
+Створіть файл з назвою `Procfile` без розширення в кореневому каталозі проєкту.
+Наприклад, `Procfile.txt` і `procfile` не є дійсними.
+Далі напишіть цей код одним рядком у такому форматі:
+
+```
+<dynos type>: <command for executing our main entry file>
```

Comment:
I think this part is safe for translation.

Reply:
```suggestion
<тип dyno>: <команда для виконання основного файлу входу>
```